test/e2e/util/util.go

```diff
@@ -1512,6 +1514,271 @@ func EnsureGuestWebhooksValidated(t *testing.T, ctx context.Context, guestClient
+	g.Expect(err).NotTo(HaveOccurred(), "failed to delete additional-pull-secret secret")
+
+	t.Run("Check if the additional-pull-secret secret is deleted in the DataPlane", func(t *testing.T) {
+		additionalPullSecret := &corev1.Secret{}
+		g.Eventually(func() error {
+			if err := guestClient.Get(ctx, client.ObjectKey{Name: additionalPullSecretName, Namespace: additionalPullSecretNamespace}, additionalPullSecret); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}
+			return fmt.Errorf("additional-pull-secret secret is still present")
+		}, 2*time.Minute, 5*time.Second).Should(Succeed(), "additional-pull-secret secret is still present")
+	})
+
+	// Check if the GlobalPullSecret secret is deleted in the DataPlane
+	t.Run("Check if the GlobalPullSecret secret is deleted in the DataPlane", func(t *testing.T) {
+		g.Eventually(func() error {
+			globalPullSecret := hccomanifests.GlobalPullSecret()
+			if err := guestClient.Get(ctx, client.ObjectKey{Name: globalPullSecret.Name, Namespace: globalPullSecret.Namespace}, globalPullSecret); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}
+			return fmt.Errorf("global-pull-secret secret is still present")
+		}, 2*time.Minute, 5*time.Second).Should(Succeed(), "global-pull-secret secret is still present")
+	})
+
+	// Check if the DaemonSet is deleted in the DataPlane
+	t.Run("Check if the DaemonSet is deleted in the DataPlane", func(t *testing.T) {
+		g.Eventually(func() error {
+			daemonSet := hccomanifests.GlobalPullSecretDaemonSet()
+			if err := guestClient.Get(ctx, client.ObjectKey{Name: daemonSet.Name, Namespace: daemonSet.Namespace}, daemonSet); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}
+			return fmt.Errorf("DaemonSet is still present")
+		}, 2*time.Minute, 5*time.Second).Should(Succeed(), "DaemonSet is still present")
+	})
+
+	// Check if the additional RBAC is deleted in the DataPlane
+	t.Run("Check if the additional RBAC is deleted in the DataPlane", func(t *testing.T) {
+		g.Eventually(func() error {
+			role := hccomanifests.GlobalPullSecretRole()
+			if err := guestClient.Get(ctx, client.ObjectKey{Name: role.Name, Namespace: role.Namespace}, role); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}
+
+			roleBinding := hccomanifests.GlobalPullSecretRoleBinding()
+			if err := guestClient.Get(ctx, client.ObjectKey{Name: roleBinding.Name, Namespace: roleBinding.Namespace}, roleBinding); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}
+
+			serviceAccount := hccomanifests.GlobalPullSecretServiceAccount()
+			if err := guestClient.Get(ctx, client.ObjectKey{Name: serviceAccount.Name, Namespace: serviceAccount.Namespace}, serviceAccount); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
+			}
+
```

Comment:
this should check the disk in file is as expected after deletion or that the image can't be pulled anymore